lib/Optimizer/Dialect/Quake/QuakeOps.cpp

```diff
@@ -382,8 +382,15 @@ LogicalResult quake::ExtractRefOp::verify() {
       return emitOpError(
           "must not have both a constant index and an index argument.");
   } else {
-    if (getRawIndex() == kDynamicIndex)
+    if (getRawIndex() == kDynamicIndex) {
       return emitOpError("invalid constant index value");
+    } else {
+      auto veqSize = getVeq().getType().getSize();
+      if (veqSize > 0 && getRawIndex() >= veqSize)
```

Comment:
Funny enough, that's how I had it done first, but I decided it was more beneficial to optimize it by removing 3 of the function calls, but you're correct that it would make a subtle dependence on the magic `0` value, so I'll back it out. Updated in 6f48dc5b.